### PR TITLE
Show all catalog entries

### PR DESCRIPTION
**Pull request by Himangini**
### Description
Added endpoint to return all catalog entries for GET `/profiles` endpoint
Issue #164 

### Checklist
- [x] Added tests that cover your change
- [ ] Added/modified documentation as required (such as the `README.md` (diagrams, usage, roadmap etc), and anything in `examples/`)
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] Added a `kind` label to the PR (e.g. `kind/feature`)

### CODE REVIEW

File: pkg/api/api_test.go [hunk 1]
```diff
@@ -90,22 +90,70 @@ var _ = Describe("Api", func() {
 				Expect(actualProfileName).To(Equal("nginx"))
 			})
 		})
+	})
 
-		When("no name query is provided", func() {
+	Context("/profiles", func() {
+		When("few profiles exist", func() {
```
Comment (Callisto13) [inline pkg/api/api_test.go:96]: would this be more accurate:
```suggestion
		When("no query string is passed to the search endpoint", func() {
```

File: pkg/api/api.go [hunk 2]
```diff
@@ -51,13 +53,15 @@ func New(profileCatalog Catalog, logger logr.Logger) API {
...
+	var result []profilesv1.ProfileCatalogEntry
 	logger := a.logger.WithValues("endpoint", r.URL.Path, "name", query)
+
 	if query == "" {
-		a.logger.Error(fmt.Errorf("missing query param"), "name param not set", "name", query)
-		a.logAndWriteHeader(w, http.StatusBadRequest)
-		return
+		result = a.profileCatalog.SearchAll()
+	} else {
+		result = a.profileCatalog.Search(query)
```
Comment (aclevername) [inline pkg/api/api.go:62]: shall we log when searching for query vs all?

File: pkg/api/api_test.go (hunk 1 above)
Comment (Himangini) [reply pkg/api/api_test.go]: done 👍🏻

File: pkg/api/api.go (hunk 2 above)
Comment (Himangini) [reply pkg/api/api.go]: added logging 👍🏻

File: pkg/api/api.go [hunk 3]
```diff
@@ -51,13 +53,17 @@ func New(profileCatalog Catalog, logger logr.Logger) API {
...
 	logger := a.logger.WithValues("endpoint", r.URL.Path, "name", query)
+
 	if query == "" {
-		a.logger.Error(fmt.Errorf("missing query param"), "name param not set", "name", query)
-		a.logAndWriteHeader(w, http.StatusBadRequest)
-		return
+		logger.Info("Searching for all available profiles")
+		result = a.profileCatalog.SearchAll()
+	} else {
+		logger.Info("Searching for profiles with the name: " + query)
```
Comment (aclevername) [inline pkg/api/api.go:63]: the logger here uses structured logging, so `logger.Info("message", key, value). So a nice refactor would be:
```suggestion
		logger.Info("Searching for profiles matching name", "name", query)
```

Comment (Himangini) [reply pkg/api/api.go]: oops my bad, fixed

**Review by aclevername — APPROVED** [review]

**Pull request merged**